addOns/ratelimit/src/main/java/org/zaproxy/addon/ratelimit/RateLimitHostMenu.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.ratelimit;
+
+import static org.zaproxy.addon.ratelimit.RateLimitParamRule.RateLimitGroupBy.HOST;
+
+import java.util.regex.Pattern;
+import org.apache.commons.httpclient.URIException;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.view.messagecontainer.http.HttpMessageContainer;
+import org.zaproxy.zap.view.popup.PopupMenuItemHttpMessageContainer;
+
+/**
+ * Content menu item to limit rate for the host.
+ *
+ * @see HttpMessageContainer
+ */
+public class RateLimitHostMenu extends PopupMenuItemHttpMessageContainer {
```

Comment:
These two menus could have a common base and then just a specific `addRule` implementation to reduce duplication...

Reply:
fixed